baseapp/abci_utils.go

```diff
@@ -86,7 +93,7 @@ func ValidateVoteExtensions(
 		}
 
 		valConsAddr := sdk.ConsAddress(vote.Validator.Address)
-		bondedTokens, cmtPubKeyProto, err := valStore.BondedTokensAndPubKeyByConsAddr(ctx, valConsAddr)
+		_, cmtPubKeyProto, err := valStore.BondedTokensAndPubKeyByConsAddr(ctx, valConsAddr)
```

Comment:
If we ignore the value, we should delete this method from x/staking. It was added recently and never have been release (see https://github.com/cosmos/cosmos-sdk/pull/17164)

Reply:
Ditto. Let's address this @davidterpay and we can merge.

Reply:
I'll some docs on why we are utilizing the voting power on `CometBFT` instead of most recent bonded tokens on base app. But at a high level, liveness of the network is dependent on the consensus power each validator has as derived by base app. But this voting power has a delay in terms of relaying to `CometBFT`. What this means is that for vote extensions to be in sync with the underlying consensus engine, they should default to voting power from the VEs themselves - not from baseapp at the current block. Otherwise you have liveness issues.

Copying this over from the open issue related to this. 

Say there are two validators in the network, val1 and val2:

Block H - 1 or H -2:
- Val1 has a voting power of 80 (from the perspective of the consensus engine CometBFT)
- Val2 has a voting power of 20 (from the perspective of the consensus engine CometBFT)

Only val1 signs off on block H - 1 which constitutes a valid block (> 2/3+). In block H-1, there is a redelegation transaction that moves 30 voting power from val1 to val2.

Block H:
- is currently being constructed and contains the signed vote extensions from Block H - 1.

Within `ValidateVoteExtensions`,
- Val1 has voting power of 50 (from the perspective of the application)
- Val2 has a voting power of 50 (from the perspective of the application)

When validator vote extensions is called in prepare/process proposal, it will see that Val1 only has a voting power of 50 and will reject the set of vote extensions (even though it should still be 80 since this is what comet is utilizing).

Reply:
Do you think this should be added to the ADR or within the code itself?

Reply:
I'd say both 😬